diff --git a/compiler/GHC/Driver/Session.hs b/compiler/GHC/Driver/Session.hs
index 06d8005..495e211 100644
--- a/compiler/GHC/Driver/Session.hs
+++ b/compiler/GHC/Driver/Session.hs
@@ -3748,6 +3748,7 @@ validHoleFitsImpliedGFlags
 default_PIC :: Platform -> [GeneralFlag]
 default_PIC platform =
   case (platformOS platform, platformArch platform) of
+    (OSLinux,   ArchX86_64)  -> [Opt_PIC]
     -- Darwin always requires PIC.  Especially on more recent macOS releases
     -- there will be a 4GB __ZEROPAGE that prevents us from using 32bit addresses
     -- while we could work around this on x86_64 (like WINE does), we won't be
