diff --git a/scripts/build-dependencies-mac.sh b/scripts/build-dependencies-mac.sh
index 50d2d3942b..3b2c5f17b0 100755
+-- a/scripts/build-dependencies-mac.sh
-++ b/scripts/build-dependencies-mac.sh
@@ -5,7 +5,7 @@ set -e
 export MACOSX_DEPLOYMENT_TARGET=11.0
 INSTALLDIR="$HOME/deps"
 NPROCS="$(getconf _NPROCESSORS_ONLN)"
+SDL=SDL2-2.28.5
-SDL=SDL2-2.30.0
 QT=6.6.0
 MOLTENVK=1.2.6
 
@@ -18,7 +18,7 @@ export CFLAGS="-I$INSTALLDIR/include -Os $CFLAGS"
 export CXXFLAGS="-I$INSTALLDIR/include -Os $CXXFLAGS"
 
 cat > SHASUMS <<EOF
+332cb37d0be20cb9541739c61f79bae5a477427d79ae85e352089afdaf6666e4  $SDL.tar.gz
-36e2e41557e0fa4a1519315c0f5958a87ccb27e25c51776beb6f1239526447b0  $SDL.tar.gz
 b6a3d179aa9c41275ed0e35e502e5e3fd347dbe5117a0435a26868b231cd6246  v$MOLTENVK.tar.gz
 039d53312acb5897a9054bd38c9ccbdab72500b71fdccdb3f4f0844b0dd39e0e  qtbase-everywhere-src-$QT.tar.xz
 e1542cb50176e237809895c6549598c08587c63703d100be54ac2d806834e384  qtimageformats-everywhere-src-$QT.tar.xz
diff --git a/scripts/build-dependencies-windows-arm64.bat b/scripts/build-dependencies-windows-arm64.bat
index beb05f63c9..0562b902c5 100644
+-- a/scripts/build-dependencies-windows-arm64.bat
-++ b/scripts/build-dependencies-windows-arm64.bat
@@ -43,9 +43,9 @@ cd "%BUILDDIR%"
 
 set QT=6.6.1
 set QTMINOR=6.6
+set SDL=SDL2-2.28.5
-set SDL=SDL2-2.30.0
 
+call :downloadfile "%SDL%.zip" "https://libsdl.org/release/%SDL%.zip" 97bd14ee0ec67494d2b93f1a4f7da2bf891103c57090d96fdcc2b019d885c76a || goto error
-call :downloadfile "%SDL%.zip" "https://libsdl.org/release/%SDL%.zip" 80b0c02b6018630cd40639ac9fc8e5c1d8eec14d8fe3e6dfa76343e3ba8b78d9 || goto error
 call :downloadfile "qtbase-everywhere-src-%QT%.zip" "https://download.qt.io/official_releases/qt/%QTMINOR%/%QT%/submodules/qtbase-everywhere-src-%QT%.zip" 818f92518d1a89ee98ae818891a7d2f0e41aa45b933d55215da2df6d5459428e || goto error
 call :downloadfile "qtimageformats-everywhere-src-%QT%.zip" "https://download.qt.io/official_releases/qt/%QTMINOR%/%QT%/submodules/qtimageformats-everywhere-src-%QT%.zip" 03f01042f86b4dbf7329a179f20835817c660a183178c11570cc0535b3c3ba58 || goto error
 call :downloadfile "qtsvg-everywhere-src-%QT%.zip" "https://download.qt.io/official_releases/qt/%QTMINOR%/%QT%/submodules/qtsvg-everywhere-src-%QT%.zip" d44d5ead8d4682f54c91687b5e32f2735f086419e3889e05609feae1a7f02da9 || goto error
diff --git a/scripts/build-dependencies-windows-x64.bat b/scripts/build-dependencies-windows-x64.bat
index 3bc14a8bdc..a41130ac08 100644
+-- a/scripts/build-dependencies-windows-x64.bat
-++ b/scripts/build-dependencies-windows-x64.bat
@@ -39,9 +39,9 @@ cd "%BUILDDIR%"
 
 set QT=6.6.1
 set QTMINOR=6.6
+set SDL=SDL2-2.28.5
-set SDL=SDL2-2.30.0
 
+call :downloadfile "%SDL%.zip" "https://libsdl.org/release/%SDL%.zip" 97bd14ee0ec67494d2b93f1a4f7da2bf891103c57090d96fdcc2b019d885c76a || goto error
-call :downloadfile "%SDL%.zip" "https://libsdl.org/release/%SDL%.zip" 80b0c02b6018630cd40639ac9fc8e5c1d8eec14d8fe3e6dfa76343e3ba8b78d9 || goto error
 call :downloadfile "qtbase-everywhere-src-%QT%.zip" "https://download.qt.io/official_releases/qt/%QTMINOR%/%QT%/submodules/qtbase-everywhere-src-%QT%.zip" 818f92518d1a89ee98ae818891a7d2f0e41aa45b933d55215da2df6d5459428e || goto error
 call :downloadfile "qtimageformats-everywhere-src-%QT%.zip" "https://download.qt.io/official_releases/qt/%QTMINOR%/%QT%/submodules/qtimageformats-everywhere-src-%QT%.zip" 03f01042f86b4dbf7329a179f20835817c660a183178c11570cc0535b3c3ba58 || goto error
 call :downloadfile "qtsvg-everywhere-src-%QT%.zip" "https://download.qt.io/official_releases/qt/%QTMINOR%/%QT%/submodules/qtsvg-everywhere-src-%QT%.zip" d44d5ead8d4682f54c91687b5e32f2735f086419e3889e05609feae1a7f02da9 || goto error
diff --git a/scripts/build-dependencies.sh b/scripts/build-dependencies.sh
index b79ca50ad9..2ebf78a69b 100755
+-- a/scripts/build-dependencies.sh
-++ b/scripts/build-dependencies.sh
@@ -4,7 +4,7 @@ set -e
 
 INSTALLDIR="$HOME/deps"
 NPROCS="$(getconf _NPROCESSORS_ONLN)"
+SDL=SDL2-2.28.5
-SDL=SDL2-2.30.0
 QT=6.6.1
 LIBBACKTRACE=ad106d5fdd5d960bd33fae1c48a351af567fd075
 
@@ -12,7 +12,7 @@ mkdir -p deps-build
 cd deps-build
 
 cat > SHASUMS <<EOF
+332cb37d0be20cb9541739c61f79bae5a477427d79ae85e352089afdaf6666e4  $SDL.tar.gz
-36e2e41557e0fa4a1519315c0f5958a87ccb27e25c51776beb6f1239526447b0  $SDL.tar.gz
 fd6f417fe9e3a071cf1424a5152d926a34c4a3c5070745470be6cf12a404ed79  $LIBBACKTRACE.zip
 450c5b4677b2fe40ed07954d7f0f40690068e80a94c9df86c2c905ccd59d02f7  qtbase-everywhere-src-$QT.tar.xz
 ac4ed08950072e375be662cfa64fdb447dd6e935cf29c56a4128d1500492188f  qtimageformats-everywhere-src-$QT.tar.xz
diff --git a/scripts/flatpak/modules/20-sdl2.json b/scripts/flatpak/modules/20-sdl2.json
index 2fe301fe75..ec1735591b 100644
+-- a/scripts/flatpak/modules/20-sdl2.json
-++ b/scripts/flatpak/modules/20-sdl2.json
@@ -28,8 +28,8 @@
   "sources": [
     {
       "type": "archive",
+      "url": "https://libsdl.org/release/SDL2-2.28.5.tar.gz",
+      "sha256": "332cb37d0be20cb9541739c61f79bae5a477427d79ae85e352089afdaf6666e4"
-      "url": "https://libsdl.org/release/SDL2-2.30.0.tar.gz",
-      "sha256": "36e2e41557e0fa4a1519315c0f5958a87ccb27e25c51776beb6f1239526447b0"
     }
   ],
   "cleanup": [
--- a/CMakeModules/DuckStationDependencies.cmake	2024-02-13 18:42:06.075965505 -0300
+++ b/CMakeModules/DuckStationDependencies.cmake	2024-02-13 18:43:00.328870470 -0300
@@ -1,5 +1,5 @@
 if(ENABLE_SDL2)
-  find_package(SDL2 2.30.0 REQUIRED)
+  find_package(SDL2 2.28.5 REQUIRED)
 endif()
 if(NOT WIN32 AND NOT ANDROID)
   find_package(CURL REQUIRED)
