diff --git a/util/meson.build b/util/meson.build
index 5cc209c..485ffa1 100644
--- a/util/meson.build
+++ b/util/meson.build
@@ -59,6 +59,6 @@ foreach util : cairo_utils
   )
 endforeach
 
-if conf.get('CAIRO_HAS_DLSYM', 0) == 1 and cc.has_header('execinfo.h')
+if conf.get('CAIRO_HAS_DLSYM', 0) == 1 and cc.has_function('alloc_stats_add', prefix: '#include <execinfo.h>')
   libmallocstats = library('malloc-stats', 'malloc-stats.c', dependencies : dl_dep)
 endif
